src/zkevm_specs/evm_circuit/execution/precompiles/ecrecover.py

```diff
@@ -0,0 +1,67 @@
+from zkevm_specs.evm_circuit.instruction import Instruction
+from zkevm_specs.evm_circuit.table import (
+    CallContextFieldTag,
+    FixedTableTag,
+    RW,
+)
+from zkevm_specs.util import FQ, Word, EcrecoverGas
+
+SECP256K1N = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141
+
+
+def ecRecover(instruction: Instruction):
+    is_success = instruction.call_context_lookup(CallContextFieldTag.IsSuccess, RW.Read)
+    address_word = instruction.call_context_lookup_word(CallContextFieldTag.CalleeAddress)
+    address = instruction.word_to_address(address_word)
+    instruction.fixed_lookup(
+        FixedTableTag.PrecompileInfo,
+        FQ(instruction.curr.execution_state),
+        address,
+        FQ(EcrecoverGas),
+    )
+
+    # Get msg_hash, signature and recovered address from aux_data
+    msg_hash: Word = instruction.curr.aux_data[0]
+    sig_v: Word = instruction.curr.aux_data[1]
+    sig_r: Word = instruction.curr.aux_data[2]
+    sig_s: Word = instruction.curr.aux_data[3]
+    recovered_addr: FQ = instruction.curr.aux_data[4]
+
+    is_recovered = FQ(instruction.is_zero(recovered_addr) != FQ(1))
+
+    # if the address is recovered, this call should be success either
+    instruction.constrain_equal(is_success, is_recovered)
```

Comment:
It seems `is_success` will alwasy be true when gas is sufficient (scroll handles precompile insufficient gas in another execution state) even the input is malformed (ref: https://github.com/ethereum/execution-specs/blob/master/src/ethereum/shanghai/vm/precompiled_contracts/ecrecover.py)

Reply:
fixed in [3ae72d0](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/495/commits/3ae72d0ae512e0752d445ac695f2b9d37d41696f)